### ISSUE TITLE
Raw/TO_JSON method

### ISSUE THREAD

**Issue by markushopkins** [gbxyz/perl-net-rdap#3]
is there a method to access the raw JSON response in the module, especially for a domain object?

**Comment by gbxyz** [comment]
I will look at adding a `TO_JSON()` method to `Net::RDAP::Base`, so you can call `JSON::encode($rdap)` and have it work. In the meantime here are some alternatives:

1\. Convert the object to a hash and then encode a reference to that hash:

```perl
my %domain = %{Net::RDAP->new->domain("example.com")};
my $json = JSON::XS->new->encode(\%domain);
```

2\. Use `Net::RDAP::Registry` with [Data::Mirror](https://metacpan.org/pod/Data::Mirror):

```
my $json = Data::Mirror::mirror_str(Net::RDAP::Registry->get_url(Net::DNS::Domain->new("example.com")));
```

**Comment by markushopkins** [comment]
Thanks for that.  Tying in with my other question you answered already, it looks like the corollary to the second option for preserving the raw registrar response would still involve iterating on the array created using the `links()` method to ID the element (e.g., `$link`) with `rel->'relative'` and then call [Data::Mirror](https://metacpan.org/pod/Data::Mirror) on that:

```perl
my $json = Data::Mirror::mirror_str($link->href);
```

Apologies for all the questions, I'm just trying to figure out whether I'm better of encoding as JSON or pulling a second response to treat as raw unless/until you implement `TO_JSON()` and making sure I understand the methods in the module helps a ton.  Thanks so much.

**Comment by gbxyz** [comment]
I've just released [Net::RDAP v0.22](https://metacpan.org/release/GBROWN/Net-RDAP-0.22) which adds the `TO_JSON()` method to `Net::RDAP::Base`.

**Comment by markushopkins** [comment]
Hey thanks a ton!
